Testnetwork running in dev container

**Describe the bug**
<!-- 
  Add a clear and concise description of what the bug is. 
-->

Setting up the FPC test network and playing with it within the FPC docker dev container does not work well.
Following the [README.md](https://github.com/hyperledger-labs/fabric-private-chaincode/tree/master/integration/test-network) fails when starting the network using the `./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0` command.

**To Reproduce**
```bash
cd $FPC_PATH/integration/test-network/fabric-samples
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.3.0 1.4.9 -s
cd $FPC_PATH/integration/test-network
./setup.sh
cd $FPC_PATH/integration/test-network/fabric-samples/test-network
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

Here spawning CA hangs as also experienced by @g2flyer. It seems that the relative paths in `docker-compose-ca.yaml` causes a problem for the docker daemon running on the host.

I added a little fix to our `setup.sh` to replace relative paths with the absolute paths based on `DOCKERD_FPC_PATH`
```bash
if [ ! -z ${DOCKERD_FPC_PATH+x} ]; then
  echo "Oo we are in docker mode! we need to use the host fpc path"
  FPC_PATH=${DOCKERD_FPC_PATH}
  FABRIC_SAMPLES=${FPC_PATH}/integration/test-network/fabric-samples
  echo "set FPC_PATH = ${FPC_PATH}"

  sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_TEST_NET}"
  sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_CA}"
fi
```

Starting from scratch ...
```bash
cd $FPC_PATH/integration/test-network/fabric-samples
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.3.0 1.4.9 -s
cd $FPC_PATH/integration/test-network
./setup.sh
cd $FPC_PATH/integration/test-network/fabric-samples/test-network
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

network starts nicely! Yah!

Let's continue ...

```bash
cd $FPC_PATH/integration/test-network
./installFPC.sh

export   ORG1_ECC_PKG_ID=echo_935E0D2570D912C3329CA710F10410844ED856BF3891063507C3EBD0FDE00954:c757bcc6a8362aa52499392f0518807c01214f8ff44f5603f6dcda85a114984a  ORG1_ERCC_PKG_ID=ercc_1.0:1e1d778324fc583c3da7950133d8e1f2d859f61a4612cd7cfaeefe885b7178d4  ORG2_ECC_PKG_ID=echo_935E0D2570D912C3329CA710F10410844ED856BF3891063507C3EBD0FDE00954:164765635539f559b8ac40ade7a9110b0ac6de726d259015b2080c128865d3ff  ORG2_ERCC_PKG_ID=ercc_1.0:136eaf3327d26564bb01208b852e8208aa5a89cfdfcd7afe9d5bf92ebf13014f
```


```bash
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network# make ercc-ecc-start
if [ "SIM" = "HW" ]; then \
        export HW_EXTENSION="-hw" \
            AESMD_PATH="/var/run/aesmd" \
            SGX_DEVICE_PATH=$(if [ -e "/dev/isgx" ]; then echo "/dev/isgx"; elif [ -e "/dev/sgx" ]; then echo "/dev/sgx"; else echo "none"; fi) && \
        [ "${SGX_DEVICE_PATH}" != "none" ] || ( echo "ERROR: SGX_MODE is HW but no sgx device found"; exit 1; ) \
fi && \
env CC_ID=echo docker-compose up
ERROR: Invalid interpolation format for "image" option in service "ecc.peer0.org1.example.com": "fpc/fpc-${CC_ID}${HW_EXTENSION:-}"
Makefile:26: recipe for target 'ercc-ecc-start' failed
make: *** [ercc-ecc-start] Error 1
```

OUCH! This command did work on Mac. It seems that docker-compose does not like the term `fpc/fpc-${CC_ID}${HW_EXTENSION:-}`, in particular, the DEFAULT mapping of `HW_EXTENSION`.

Dev container:
```bash
root@docker-desktop:/project/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/test-network# docker-compose version
docker-compose version 1.17.1, build unknown
docker-py version: 2.5.1
CPython version: 2.7.17
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
```

So with the dev container based on Ubuntu 18.04 we have a pretty old version of docker-compose.

Mac:
```bash
➜  test-network git:(master) ✗ docker-compose version
docker-compose version 1.26.2, build eefe0d31
docker-py version: 4.2.2
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

Switching to Ubuntu 20.04 to see of if this works better.

Dev container
```bash
docker-compose version 1.25.0, build unknown
docker-py version: 4.1.0
CPython version: 3.8.5
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
```

Uhh this looks much newer. Let's try again from the beginning ....

```bash
cd $FPC_PATH/integration/test-network/fabric-samples
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.3.0 1.4.9 -s
cd $FPC_PATH/integration/test-network
./setup.sh
```

It seems `yq` has not been upgraded to v4 with Ubuntu 20.04 :/


```bash
cd $FPC_PATH/integration/test-network/fabric-samples/test-network
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

Starting the network also sometime fails here. It seems the network script sometimes try to create the channel before the peers are up.
What helps is:
```bash
./network.sh up
./network.sh up createChannel -c mychannel -ca -cai 1.4.9 -i 2.3.0
```

Super wired! Let's continue.

```bash
cd $FPC_PATH/integration/test-network
./installFPC.sh
export \
  ORG1_ECC_PKG_ID=echo_3CB9C78598AE52C4E97902D1C2002D288FD0099AEB8F248DE336E50074682B44:3ad57c03ddb09f811efcb71f44e7408668565cc9fe1ed97461a807feef728a26\
  ORG1_ERCC_PKG_ID=ercc_1.0:0035786ed959f875024fd70f1ac5a7568f7662e9d8dc78078c772d45940710dd\
  ORG2_ECC_PKG_ID=echo_3CB9C78598AE52C4E97902D1C2002D288FD0099AEB8F248DE336E50074682B44:cbb5ec871b697eee3fece8ac10f2e608bf1a3b61406748ce60cf7431a83036f9\
  ORG2_ERCC_PKG_ID=ercc_1.0:59f39f2f168d630688f87ffa8bbb7bef1d0aacb23cbef1ddcd77422b81f9d6da

make ecc-container
make ercc-container
make ercc-ecc-start
```

Yah! It starts ... no docker-compose issue anymore. My conclusion: Let's switch to Ubuntu 20.04 as default.


Running the test-network on a Mac works nicely IF the correct version of `yq` is installed. See related #519 issue.

--- a/README.md
+++ b/README.md
@@ -306,6 +306,10 @@ Make sure that you have the following required dependencies installed:
 
 * Docker 18.x and docker-compose (1.25.x or higher)
 
+* yq (version 3.x. Newer versions (v4.x and higher) are currently *not* supported!)
+  You can install `yq` v3 via `go get`.
+  ```bash
+  GO111MODULE=on go get github.com/mikefarah/yq/v3
 * Protocol Buffers
     - Protocol Buffers 3.0.x needed for the Intel SGX SDK
     - Protocol Buffers 3.11.x or higher and [Nanopb](http://github.com/nanopb/nanopb) 0.4.3

--- a/README.md
+++ b/README.md
@@ -304,7 +304,7 @@ Make sure that you have the following required dependencies installed:
 
 * [Go](https://golang.org/) 1.15.4 or higher
 
-* Docker 18.x
+* Docker 18.x and docker-compose (1.25.x or higher)
 
 * Protocol Buffers
     - Protocol Buffers 3.0.x needed for the Intel SGX SDK

--- a/integration/test-network/README.md
+++ b/integration/test-network/README.md
@@ -1,29 +1,35 @@
 # Setup the FPC test network
 
 The FPC test network builds on the test-network provided by [fabric-samples](https://github.com/hyperledger/fabric-samples).
-
 We provide fabric-samples as a submodule in `$FPC_PATH/integration/test-network/fabric-samples`.
 
-Make sure you have installed [yq](https://github.com/mikefarah/yq).
-Note that you will version v3.4.1 or larger. 
-For Ubuntu, `sudo snap install yq` is the easiest way to get a good version.
-
-Note if you run the FPC test network from within our FPC docker dev container, please
-use Ubuntu 20.04 Docker images by setting the corresponding settinngs in your `config.override.mk`
-before you build the dev container with `cd utils/dockder && make dev && make run`.
-```Makefile
-DOCKER_BUILD_OPTS=--build-arg UBUNTU_VERSION=20.04 --build-arg UBUNTU_NAME=focal
+In order to use FPC with the test network, make sure you have installed [yq](https://github.com/mikefarah/yq) (version 3.x).
+Not that newer versions (v4.x and higher) are currently not supported.
+You can install `yq` v3 via `go get`.
+```bash
+GO111MODULE=on go get github.com/mikefarah/yq/v3
 ```
+
+In addition to `yq` you need a recent version of docker-compose (version 1.25 or higher).
+Note Ubuntu 18.04 comes with an older version of docker-compose and thus needs to be updated.
+See related notes in [Working from behind a proxy](../../README.md#working-from-behind-a-proxy) in our [README.md](../../README.md) for more information.
+
+[comment]: <> (This comment can be removed with upgrading FPC to general Ubuntu 20.04 support)
+If you run the FPC test network from within our FPC docker dev container, please use Ubuntu 20.04 Docker images.
+```Makefile
+DOCKER_BUILD_OPTS="--build-arg UBUNTU_VERSION=20.04 --build-arg UBUNTU_NAME=focal" make -C $FPC_PATH/utils/docker run
+```
+
+## Prepare FPC containers and network
 
 Before you start the network make sure you build ercc and ecc containers:
 
 ```bash
+cd $FPC_PATH/integration/test-network/
 make build
 ```
 If you want to build with mock-ecc rather than the real enclave-based one, build with
 `make build GOTAGS="-tags mock_ecc"` instead.
-
-## Prepare network
 
 Setup fabric sample network, binaries and docker images (this follow the [instructions](https://hyperledger-fabric.readthedocs.io/en/latest/install.html)).
 

--- a/integration/test-network/README.md
+++ b/integration/test-network/README.md
@@ -9,17 +9,6 @@ You can install `yq` v3 via `go get`.
 ```bash
 GO111MODULE=on go get github.com/mikefarah/yq/v3
 ```
-
-In addition to `yq` you need a recent version of docker-compose (version 1.25 or higher).
-Note Ubuntu 18.04 comes with an older version of docker-compose and thus needs to be updated.
-See related notes in [Working from behind a proxy](../../README.md#working-from-behind-a-proxy) in our [README.md](../../README.md) for more information.
-
-[comment]: <> (This comment can be removed with upgrading FPC to general Ubuntu 20.04 support)
-If you run the FPC test network from within our FPC docker dev container, please use Ubuntu 20.04 Docker images.
-```Makefile
-DOCKER_BUILD_OPTS="--build-arg UBUNTU_VERSION=20.04 --build-arg UBUNTU_NAME=focal" make -C $FPC_PATH/utils/docker run
-```
-
 ## Prepare FPC containers and network
 
 Before you start the network make sure you build ercc and ecc containers:

--- a/integration/test-network/README.md
+++ b/integration/test-network/README.md
@@ -8,6 +8,13 @@ Make sure you have installed [yq](https://github.com/mikefarah/yq).
 Note that you will version v3.4.1 or larger. 
 For Ubuntu, `sudo snap install yq` is the easiest way to get a good version.
 
+Note if you run the FPC test network from within our FPC docker dev container, please
+use Ubuntu 20.04 Docker images by setting the corresponding settinngs in your `config.override.mk`
+before you build the dev container with `cd utils/dockder && make dev && make run`.
+```Makefile
+DOCKER_BUILD_OPTS=--build-arg UBUNTU_VERSION=20.04 --build-arg UBUNTU_NAME=focal
+```
+
 Before you start the network make sure you build ercc and ecc containers:
 
 ```bash

--- a/integration/test-network/setup.sh
+++ b/integration/test-network/setup.sh
@@ -23,7 +23,7 @@ backup() {
   fi
 }
 
-FABRIC_SAMPLES=${FPC_PATH}/integration/test-network/fabric-samples/
+FABRIC_SAMPLES=${FPC_PATH}/integration/test-network/fabric-samples
 
 if [ ! -d "${FABRIC_SAMPLES}/bin" ]; then
   echo "Error: environment not properly setup, see README.md"
@@ -35,7 +35,9 @@ fi
 # - config and docker-compose files (for fpc lite enablement)
 
 CORE_PATH=${FABRIC_SAMPLES}/config/core.yaml
-DOCKER_PATH=${FABRIC_SAMPLES}/test-network/docker/docker-compose-test-net.yaml
+DOCKER_PATH=${FABRIC_SAMPLES}/test-network/docker
+DOCKER_COMPOSE_TEST_NET=${DOCKER_PATH}/docker-compose-test-net.yaml
+DOCKER_COMPOSE_CA=${DOCKER_PATH}/docker-compose-ca.yaml
 
 # TODO the current `setup.sh` has an issue that it is not idempotent
 # In particular, when starting nodes using the ./network up ... script won't work if
@@ -45,7 +47,8 @@ backup ${CORE_PATH}
 
 yq m -i -a=append ${CORE_PATH} core_ext.yaml
 
-backup ${DOCKER_PATH}
+backup ${DOCKER_COMPOSE_TEST_NET}
+backup ${DOCKER_COMPOSE_CA}
 
 peers=("peer0.org1.example.com" "peer0.org2.example.com")
 
@@ -61,11 +64,16 @@ peers=("peer0.org1.example.com" "peer0.org2.example.com")
 if [ ! -z ${DOCKERD_FPC_PATH+x} ]; then
   echo "Oo we are in docker mode! we need to use the host fpc path"
   FPC_PATH=${DOCKERD_FPC_PATH}
-  FABRIC_SAMPLES=${FPC_PATH}/integration/test-network/fabric-samples/
+  FABRIC_SAMPLES=${FPC_PATH}/integration/test-network/fabric-samples
   echo "set FPC_PATH = ${FPC_PATH}"
+
+  sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_TEST_NET}"
+  sed -i "s+\.\./+${FABRIC_SAMPLES}/test-network/+g" "${DOCKER_COMPOSE_CA}"
 fi
 
 for p in "${peers[@]}"; do
-  yq w -i ${DOCKER_PATH} "services.\"$p\".volumes[+]" "${FPC_PATH}:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode"
-  yq w -i ${DOCKER_PATH} "services.\"$p\".volumes[+]" "${FABRIC_SAMPLES}/config/core.yaml:/etc/hyperledger/fabric/core.yaml"
+  yq w -i ${DOCKER_COMPOSE_TEST_NET} "services.\"$p\".volumes[+]" "${FPC_PATH}:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode"
+  yq w -i ${DOCKER_COMPOSE_TEST_NET} "services.\"$p\".volumes[+]" "${FABRIC_SAMPLES}/config/core.yaml:/etc/hyperledger/fabric/core.yaml"
 done
+
+

